### ISSUE TITLE
Testng skips all test classes from suite

### ISSUE THREAD

**Issue by priyanshus** [cbeust/testng#739]
If you have two @BeforeClass in a test class and where second @BeforeClass depends on first one like this:

```
@BeforeClass
public void doLogin() {// Do Login}

@BeforeClass(dependsOnMethods = "doLogin")
public void logout() {// Do logout}
```

Then it will skip all the test classes from suite, if for any reason first @BeforeClass fails or throws exception.

**Comment by dbrusilovsky** [comment]
Does alwaysRun not prevent this problem?

**Comment by juherr** [comment]
@priyanshus Could you share us a little project that show the issue?

**Comment by priyanshus** [comment]
@dbrusilovsky No it does not work with alwaysRun as well.
@juherr There is nothing special in my project, its just a testng suite which contains few test classes. One of test class has two @BeforeClass methods and second @BeforeClass depends on first one as I mentioned in my first comment. 

If first @BeforeClass fails due to any reason then it skip all other test classes which are supposed to be run after failing class.

**Comment by juherr** [comment]
@priyanshus I can't reproduce your issue. Could you check my use case and tell me if I'm missing something?

@cbeust The `*Invocation` methods of `IInvokedMethodListener` are notified for "testA" which is never invoked. IMO it shouldn't. What do you think?

**Comment by priyanshus** [comment]
@juherr Please check my comment in use case of #739.

**Comment by juherr** [comment]
@priyanshus I've just updated the use case like your comment but it is working as expected. See 3d7de5c

**Comment by priyanshus** [comment]
@juherr 
Please check my inline comment in 3d7de5c.

**Comment by juherr** [comment]
@priyanshus Reproducd and fixed in #742

**Comment by piyush-mehta** [comment]
Hi, i am also getting the same issue while using testng. Following is the structure:

Base Class having:
@BeforeClass
@BeforeMethod
@AfterClass
@AfterMethod

Test Class extends BaseClass having
- @Test 

can anyone help, Thanks !!!

**Comment by juherr** [comment]
@piyush-mehta Which TestNG version do you use?

BTW, could share a more complete example we are able to run?

**Comment by mrunalimohane** [comment]
Using testng 6.11, is this still an issue?

**Comment by juherr** [comment]
@mrunalimohane It should be fixed since 6.9.5

**Comment by mrunalimohane** [comment]
Seems like it worked all fine untill version 6.10. With 6.11 the issue has come back.

**Comment by krmahadevan** [comment]
@mrunalimohane - Do you have a sample that can be used to recreate the problem ? That would be easier to start debugging.

**Comment by nirmitsrivastava** [comment]
Hello @krmahadevan @juherr
Using 6.11 Testng and any exception in BeforeClass skips the remaining test cases from other classes too.
Please check and let me know if you need more information regarding the issue.

**Comment by krmahadevan** [comment]
@nirmitsrivastava - Please help do the following:

1. Retry the same scenario using the latest released version of TestNG ( 7.0.0-beta1 )
2. If the problem persists, please create a new issue and add a simple standalone sample that can be used to recreate the problem.
